tiledb/sm/config/config.cc

```diff
@@ -924,39 +937,60 @@ const char* Config::get_from_config(
   return *found ? it->second.c_str() : "";
 }
 
-const char* Config::get_from_config_or_env(
+const char* Config::get_from_config_or_fallback(
     const std::string& param, bool* found) const {
-  // First let's see if the User has set the parameter
-  // If it is not a user set parameter it might be a default value if found in
-  // the config
+  // First check if the user has set the parameter
+  // If not, it may be a default value (if found in the config)
   bool user_set_parameter = set_params_.find(param) != set_params_.end();
 
-  // First check config
+  // [1. user-set config parameters]
   bool found_config;
   const char* value_config = get_from_config(param, &found_config);
-  // If its a user set parameter from the config return it
   if (found_config && user_set_parameter) {
     *found = found_config;
     return value_config;
   }
 
-  // Check if default for this parameter should be ignored based on
-  // environment variables
+  // Check if param default should be ignored based on environment variables
   if (ignore_default_via_env(param)) {
     *found = true;
     return "";
   }
 
-  // Check env if not found in config or if it was found in the config but is
-  // a default value
+  // [2. env variables]
   const char* value_env = get_from_env(param, found);
   if (*found)
     return value_env;
 
-  // At this point the value was not found to be user set in the config or an
-  // environmental variable so return any default value from the config or
-  // indicate it was not found
+  // [3. user-set profiles]
+  auto profile = RestProfile::load_if_exists(
+      param_values_.at("profile_name"), param_values_.at("profile_homedir"));
+  if (profile.has_value()) {
+    // The "s3.verify_ssl" parameter _may or may not_ be set on the profile.
+    // If that's the param to be fetched, see if it's set on the profile,
+    // and return its value (if it has one).
+    if (strcmp(param.c_str(), "s3.verify_ssl") == 0) {
+      auto verify_ssl = profile->get_verify_ssl();
+      if (verify_ssl.has_value()) {
+        *found = true;
+        return verify_ssl.value() ? "true" : "false";
+      }
+    } else {
+      // Fetch all other params from the profile normally.
+      try {
+        const char* value = std::move(profile->get_param(param).c_str());
```

Comment:
The `std::move` here is moving a `const char *` right?  That's a no-op but it makes me think way harder about this line of code than I would like.

Reply:
I agree. Any suggestions on this?